fuzz/generate_proto_fuzz.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Regenerates fuzz tests for boilerplate protocol tests by looking at the
+`proto_fuzz.txt` file, which is a file containing newline-delimited Rust
+paths, which refer to implementations of `manticore::protocol::Command`.
+
+This script exists to minimize the boilerplate of doing so, seeing as such
+targets tend to be quite simple.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+import re
+from pathlib import Path
+
+REPO_TOP = Path(__file__).parent.parent
+
+FUZZ_DIR = REPO_TOP / 'fuzz'
+FUZZ_CONFIG = FUZZ_DIR / 'proto_types.txt'
+FUZZ_TEMPLATES = FUZZ_DIR / 'templates'
+FUZZ_GEN = FUZZ_DIR / 'gen'
+
+FUZZ_CARGO = FUZZ_DIR / 'Cargo.toml'
+CARGO_TPL = FUZZ_TEMPLATES / 'target.toml.tpl'
+CARGO_END = '## BEGIN GENERATED TARGETS'
+
+FUZZ_YML = REPO_TOP / '.github/workflows/fuzz.yml'
+YML_TPL = FUZZ_TEMPLATES / 'fuzz_job.yml.tpl'
+YML_END = '  ## BEGIN GENERATED JOBS'
+
+def eprint(*args):
+  "Prints to stderr."
+  print(*args, file = sys.stderr)
+
+def cargo_fuzz_build():
+  out = subprocess.run(
+      ['cargo', '+nightly', '--quiet', 'fuzz', 'build'],
+      cwd = str(FUZZ_DIR))
+  eprint()
+  eprint(f'Cargo exited with {out.returncode}')
+  if out.returncode != 0:
+    eprint(f'Fuzz build failed!')
+    sys.exit(out.returncode)
+
+def reset_generated_config(path, delim):
+  text = path.read_text()
+  return text[:text.index(delim)] + delim + '\n'
+
+def main():
+  command = ' '.join(sys.argv)
+
+  eprint(f'Regenerating {FUZZ_GEN}')
+  shutil.rmtree(str(FUZZ_GEN))
+  FUZZ_GEN.mkdir()
+
+  cargo_toml = reset_generated_config(FUZZ_CARGO, CARGO_END)
+  cargo_tpl = CARGO_TPL.read_text()
+
+  fuzz_yml = reset_generated_config(FUZZ_YML, YML_END)
+  yml_tpl = YML_TPL.read_text()
+
+  count = 0
+  for ty in FUZZ_CONFIG.read_text().split('\n'):
+    if not ty or ty.startswith('#'): continue
```

Comment:
nit: continue on next line? (here and below)

Reply:
I liked the compactness but sure. =P